Readme
RVSAO -- An IRAF package to obtain radial velocities from spectra
 
The package RVSAO defined in this directory was developed by Jessica Mink at
the Harvard-Smithsonian Center for Astrophysics from the redshift package
originally written by Gerard Kriss at Johns Hopkins University and modified
significantly by Steve Levine at the University of Wisconsin.  It obtains
radial velocities and velocity dispersions using cross-correlation methods
or emission line fits.  It consists of several SPP tasks:
XCSAO and PXCSAO to cross-correlate spectra,
EMSAO and PEMSAO to find emission lines in spectra,
BCVCORR to compute solar system barycentric velocity correction,
SUMSPEC to add and/or modify spectra,
LINESPEC to creat synthetic emission line templates,
EQWIDTH to compute equivalent widths of lines in spectra,
WLRANGE Return the wavelength overlap range for a list of spectra
LISTSPEC List pixel, wavelength, delta wavelength, and/or pixel value for spectrum image
PIX2WL Compute the wavelength at a given pixel in a spectrum
WL2PIX Compute the pixel at a given wavelength in a spectrum
VELSET Artificially change the redshift of (log-wavelength) spectra
plus several CL scripts.

If you have any problems, please contact Jessica Mink, jmink@cfa.harvard.edu.

A task, RELEARN, has been provided to aid in updating parameters.  Run it
instead of UNLEARN to keep your current parameters settings while adding
new parameters.  It can fail when parameters are dropped; just re-run it
until it works, and things will be OK.

A set of templates commonly used at the CfA is included in the Templates/
subdirectory.  Someday, there will be documentation about them.

After the Revision Notes, you will find installation instructions.

A fairly complete description of this package has been published in the
August 1998 issue of the Publications of the Astronomical Society of the
Pacific:
http://adsabs.harvard.edu/cgi-bin/nph-bib_query?bibcode=1998PASP..110..934K

For later publications, see http://tdc-www.harvard.edu/iraf/rvsao/

***************************************************************************

RVSAO Revision Log

2.8.2, May 2, 2014
* Edit cosmic rays out of a spectrum by resetting a point at a time
* (using the , key) or by linearly interpolating between points on
* either side of a cosmic ray (using the d key twice).  The edited
* spectrum is automatically written out if you are running qplot or
* are saving information to the spectrum header and leave with "q".
XCSAO,EMSAO,PXCSAO,PEMSAO: Write back edited spectrum
xcfit.x: Call putspec to update spectrum after editing it (2014-05-02)
emplot.x: Fix bug which reversed red and blue limits (2013-03-15)
emplot.x: Add , command to replace individual pixel values (2014-05-01)
emplot.x: Add > command to write out edited spectrum (2014-05-02)

2.8.1, May 25, 2012
PEMSAO: Add maxlength parameter to set output spectrum file length in mode 11
xcrslts.x: add parameter maxlength to allow longer filename in mode 11

2.8.0, April 13, 2012
LISTSPEC: Add velform format for velocity per pixel output
SKYPLOT: Base on PEMSAO instead of XCSAO, like EMPLOT
juldate.x: Do not recompute HJD if already in header
juldate.x: Check for BJD as well as HJD
vcombine.x: Do not add constant dispersion error of 15 km/sec any more
xcfile.x: Only include file name in xc file if given full pathname
xcfile.x: Drop ".ms.fits" and ".fits" from object and template filenames
xcfile.x: Add ".xcor" file extension to output filename
xcfile.x: Fix bug by redimensioning title from SZ_LINE to SZ_PATHNAME
Add new shell script rvsaopar to set RVSAO task parameters from a dpar file
Add new shell script pxcsao to execute PXCSAO

2.7.9, August 11, 2010
PEMSAO: Write median (MEDPIX) and mean (MEANPIX) pixel shift to image header
PEMSAO: Do not write back to parameter file if save_vel=yes
PEMSAO: Skip apertures where no lines are found for means and medians
PEMSAO and PXCAO: Set FITS extension with specext keyword
PXCSAO and XCSAO: initialize all pointers to NULL; free only if not NULL
xcfit.x: Return properly if there is an error reading a spectrum
getspec.x: Return error if spectrum has no non-zero pixels
getimage.x:  If stacked spectra and CD2_2 is not set, set it to 1
getimage.x:  Always add extension to file name, even if it is zero

2.7.8, November 20, 2009
XCSAO, PXCSAO, EMSAO, PEMSAO: Zoom and line identification changes in emplot.x
LISTSPEC,PIX2WL,WL2PIX: Increase image string size to SZ_PATHNAME(511) from SZ_FNAME(255)
emplot.x: Fix zoom at left and right edges of a displayed spectrum.
          Replot immediately with new velocity if line selected
          Mark VELOCITY as being used if it is guessed from a line
          Change to correlation velocity automatically if switching templates
emplot.x: zoom and line selection
getimage.x: Increase image string size to SZ_PATHNAME(511) from SZ_FNAME(255)

2.7.7, July 16, 2009
XCSAO: If DIVCONT is T in object spectrum, divide it by its continuum and
       subtract 1.0 instead of subtracting the continuum; do the same with
       all templates, if they have no SUBCONT flag.
XCSAO,EMSAO: If contspec.order is zero, do not remove continuum
XCSAO,EMSAO,SUMSPEC: If dividing continuum, normalize by mean of continuum
       to get the true variation in count rates.
EQWIDTH: Compute sigma according to John Huchra's new formula

2.7.6, June 15, 2009
WLRANGE: New task to return the overlapping wavelength range for a list of spect
ra

2.7.5, June 10, 2009
EQWIDTH: Major changes to make it more robust and handle rest wavelengths right
EMSAO, PEMSAO: Fix multiple aperture plotting so it doesn't crash or print blank
 lines
PEMSAO: Compute lambda/pix by offset from line center to avoid dividing by
        zero if line is exactly at rest wavelength (as for a sky or
        calibration lamp line)
SUMSPEC: Fix bugs so that rebinned wavelength range can be computed from
         overlap
emrslts.x: Return immediately if no log file is set
emplot.x: Get number of apertures from im structure, not smw
emlines.dat: Add Lyman Beta and Gamma; move Lyamn Alpha from 1215.7 to 1215.67
             Fix name of 1240.0 from LyA to NV

2.7.4, April 24, 2009
BCFCORR: Add SUBGRAV parameter to make solar gravitational redshift subtraction
         from twilight sky spectra optional
LISTSPEC: Allow multiple apertures and write them to separate files
EQWIDTH: Fix error computation for equivalent widths and indices
SUMSPEC: Handle spectra with PIXEL WCS correctly
getimage.x: Set RA, DEC, ST, HA to 0:00 if not in header [for shdr_open()]
juldate.x: Add MJD as possible JD header keyword; return hjd = jd if
           there is no pointing direction in the image header

2.7.3, March 31, 2009
Add new shell scripts in Shell/ to run most RVSAO tasks on the Unix/Linux
command line.

2.7.2, March 13, 2009
XCSAO: Add option to shift orders when cross-correlating echelle spectra

2.7.1, February 20, 2009
EMSAO,XCSAO: Assume RA is in degrees if decimal keyword value, not hours

2.7.0, January 30, 2009
SUMSPEC: Fix bug which put back heliocentric velocity correction when shifting
         to a specific velocity and delete IRAF WCS keywords if one spectrum
         is being rebinned.
EQWIDTH: Fix error of index so that it is at least closer to correct.
         Fix bug so that a lack of a VELOCITY or BCV keyword does not crash
         the program.  Clean up exposure scaling.

See the NEWS file for the earlier history of the software.

***************************************************************************

To install this package in your local IRAF system, take the following steps:
 
1)	The package is distributed as a tar archive; IRAF is distributed
	with a tar reader.  The tar archive may be obtained by magnetic
	tape or anonymous ftp.  For magnetic tape go to step [2] and when
	reading the tar archive simply mount the tape and use the tape
	device name for the archive name in step [4].  To obtain the package
	via ftp (assuming a UNIX computer):

	    % ftp cfa-ftp.harvard.edu
	    login: anonymous
	    password: [your user name]
	    ftp> cd pub/iraf
	    ftp> get rvsao-2.8.0.readme
	    ftp> binary
	    ftp> get rvsao-2.8.0.tar.gz
	    ftp> quit
	    % ls -l
	    -rw-r--r--  1 mink          xxxx xxx xx xx:xx rvsao-2.8.0.readme
	    -rw-r--r--  1 mink        xxxxxx xxx xx xx:xx rvsao-2.8.0.tar.gz
	    %

2)	Create a directory to contain the RVSAO external package files.  This
	directory should be outside the IRAF directory tree and must be owned
	by the IRAF account.  In the following examples, this root directory
	is /local/rvsao/.  Make the appropriate file name substitutions for
        your site.
 
3)	Log in as IRAF and edit the extern.pkg file in the hlib directory to
	define the package to the CL.  From the IRAF account, outside the CL,
	you can move to this directory with the command:

            % cd $hlib

	Define the environment variable rvsao to be the pathname to the rvsao
	root directory.  UNIX pathnames must be terminated with a '/'.
	Edit extern.pkg to include:

            reset rvsao	= <directory>/rvsao-2.8.0/
            task  rvsao.pkg   = rvsao$rvsao.cl

	or, to avoid conflicts with earlier versions or the IRAF rv package,

            reset r2rvsao	= <directory>/rvsao-2.8.0/
            task  r2rvsao.pkg   = r2rvsao$r2rvsao.cl
 
	Near the end of the hlib$extern.pkg file, update the definition
	of helpdb so it includes the rvsao help database, copying the syntax
	already used in the string.  Add this line before the line containing
	a closing quote:
 
	    ,rvsao$lib/helpdb.mip\

	or, if using r2rvsao,
 
	    ,r2rvsao$lib/helpdb2.mip\
 
4)      Add to hlib$clpackage.men the package description:
 
            rvsao - Radial velocity package from SAO
 
	or, if using r2rvsao,
 
            r2rvsao - Radial velocity package from SAO
 
5)	Unpack the tar file into the RVSAO root directory.

	On a UNIX system, where ftpdir is the pathname of the directory
	into which the package was ftp'ed and rvsao is the RVSAO root
	directory:
	    % cd $rvsao (or $r2rvsao)
	    % cd ..
	    % zcat $ftpdir/rvsao-2.8.0.tar.gz | tar xvf -

	The archive file can be deleted once the package has been
	successfully installed.
 
6)      Restart IRAF and "cd rvsao" or "cd r2rvsao".

7)	Symbolically link the appropriate binary directory to bin.
	bin is originally linked to bin.generic for distribution.

        On a SPARCstation running Solaris, type
	    cl> mkpkg ssun

        On a PC running Redhat Linux, type
	    cl> mkpkg redhat

        On an Apple Macintosh running OS X on an Intel processor, type
	    cl> mkpkg macintel

        On an Apple Macintosh running OS X, type
	    cl> mkpkg macosx

        On a PC running Cygwin, type
	    cl> mkpkg cygwin

        On a PC running Linux, type
	    cl> mkpkg linux

	For other architectures, make a directory of the proper bin.{arch}
	name, and link it to bin

8)      Make the package by typing
	    cl> mkpkg update >&spool

	The spool file should be reviewed upon completion to make sure
	there were no errors.

9)	Up-to-date hypertext help is available on the World Wide Web at

	    http://tdc-www.harvard.edu/iraf/rvsao/

-Jessica Mink, SAO Telescope Data Center, 2012-05-25


NEWS
RVSAO Revision Log

2.8.2, May 2, 2014
* Edit cosmic rays out of a spectrum by resetting a point at a time
* (using the , key) or by linearly interpolating between points on 
* either side of a cosmic ray (using the d key twice).  The edited 
* spectrum is automatically written out if you are running qplot or 
* are saving information to the spectrum header and leave with "q".
XCSAO,EMSAO,PXCSAO,PEMSAO: Write back edited spectrum
xcfit.x: Call putspec to update spectrum after editing it (2014-05-02)
emplot.x: Fix bug which reversed red and blue limits (2013-03-15)
emplot.x: Add , command to replace individual pixel values (2014-05-01)
emplot.x: Add > command to write out edited spectrum (2014-05-02)

2.8.1, May 25, 2012
PXCSAO: Add maxlength parameter to set output spectrum file length in mode 11
xcrslts.x: add parameter maxlength to allow longer filename in mode 11

2.8.0, April 13, 2012
LISTSPEC: Add velform format for velocity per pixel output
SKYPLOT: Base on PEMSAO instead of XCSAO, like EMPLOT
juldate.x: Do not recompute HJD if already in header
juldate.x: Check for BJD as well as HJD
vcombine.x: Do not add constant dispersion error of 15 km/sec any more
xcfile.x: Only include file name in xc file if given full pathname
xcfile.x: Drop ".ms.fits" and ".fits" from object and template filenames
xcfile.x: Add ".xcor" file extension to output filename
xcfile.x: Fix bug by redimensioning title from SZ_LINE to SZ_PATHNAME
Add new shell script rvsaopar to set RVSAO task parameters from a dpar file
Add new shell script pxcsao to execute PXCSAO

2.7.9, August 11, 2010
PEMSAO: Write median (MEDPIX) and mean (MEANPIX) pixel shift to image header
PEMSAO: Do not write back to parameter file if save_vel=yes
PEMSAO: Skip apertures where no lines are found for means and medians
PEMSAO and PXCAO: Set FITS extension with specext keyword
PXCSAO and XCSAO: initialize all pointers to NULL; free only if not NULL
xcfit.x: Return properly if there is an error reading a spectrum
getspec.x: Return error if spectrum has no non-zero pixels
getimage.x:  If stacked spectra and CD2_2 is not set, set it to 1
getimage.x:  Always add extension to file name, even if it is zero

2.7.8, November 20, 2009
XCSAO, PXCSAO, EMSAO, PEMSAO: Zoom and line identification changes in emplot.x
LISTSPEC,PIX2WL,WL2PIX: Increase image string size to SZ_PATHNAME(511) from SZ_FNAME(255)
emplot.x: Fix zoom at left and right edges of a displayed spectrum.
          Replot immediately with new velocity if line selected
          Mark VELOCITY as being used if it is guessed from a line
          Change to correlation velocity automatically if switching templates
getimage.x: Increase image string size to SZ_PATHNAME(511) from SZ_FNAME(255)

2.7.7, July 28, 2009
XCSAO: If DIVCONT is T in object spectrum, divide it by its continuum and
       subtract 1.0 instead of subtracting the continuum; do the same with
       all templates, if they have no SUBCONT flag.
XCSAP,PXCSAO: Increase maximum number of point sin peak fir from 512 to 1024 for TRES
XCSAO,EMSAO: If contspec.order is zero, do not remove continuum
XCSAO,EMSAO,SUMSPEC: If dividing continuum, normalize by mean of continuum
       to get the true variation in count rates.
EQWIDTH: Compute sigma according to John Huchra's new formula

2.7.6, June 15, 2009
WLRANGE: New task to return the overlapping wavelength range for a list of spectra

2.7.5, June 10, 2009
EQWIDTH: Major changes to make it more robust and handle rest wavelengths right
EMSAO, PEMSAO: Fix multiple aperture plotting so it doesn't crash or print blank lines
PEMSAO: Compute lambda/pix by offset from line center to avoid dividing by
        zero if line is exactly at rest wavelength (as for a sky or
        calibration lamp line)
SUMSPEC: Fix bugs so that rebinned wavelength range can be computed from
         overlap
emrslts.x: Return immediately if no log file is set
emplot.x: Get number of apertures from im structure, not smw
emlines.dat: Add Lyman Beta and Gamma; move Lyamn Alpha from 1215.7 to 1215.67
             Fix name of 1240.0 from LyA to NV

2.7.4, April 24, 2009
BCFCORR: Add SUBGRAV parameter to make solar gravitational redshift subtraction
         from twilight sky spectra optional
LISTSPEC: Allow multiple apertures and write them to separate files
EQWIDTH: Fix error computation for equivalent widths and indices
SUMSPEC: Handle spectra with PIXEL WCS correctly
getimage.x: Set RA, DEC, ST, HA to 0:00 if not in header [for shdr_open()]
juldate.x: Add MJD as possible JD header keyword; return hjd = jd if
           there is no pointing direction in the image header

2.7.3, March 31, 2009
Add new shell scripts in Shell/ to run most RVSAO tasks on the Unix/Linux
command line.

2.7.2, March 13, 2009
XCSAO: Add option to shift orders when cross-correlating echelle spectra

2.7.1, February 20, 2009
EMSAO,XCSAO: Assume RA is in degrees if decimal keyword value, not hours

2.7.0, January 30, 2009
SUMSPEC: Fix bug which put back heliocentric velocity correction when shifting
         to a specific velocity and delete IRAF WCS keywords if one spectrum
         is being rebinned.
EQWIDTH: Fix error of index so that it is at least closer to correct.
         Fix bug so that a lack of a VELOCITY or BCV keyword does not crash
         the program.  Clean up exposure scaling.

2.6.9, December 9, 2008
EQWIDTH: Correctly compute Index and Index Error; Set to 0.0 if no continuum
EMSAO,EMPLOT: Plot line name and wavelength if only labeling emission lines
Add emlineshe2.dat He2 line list

2.6.8, October 7, 2008
EMSAO,EMPLOT: Add ) and ( commands to move through orders or apertures

2.6.7, September 16, 2008
RVRELEARN: New task to relearn all of the parameters from all of the tasks
           in RVSAO.  Old parameters keep their values; new parameters will
           be assigned default values.
SUMSPEC: Keep pointing direction if numbers of input and output apertures are 
         the same.

2.6.6, September 11, 2008

Edspec/rvsao.h
# File lib/rvsao.h
# March 15, 2013

# RVSAO 2.8.2 header file

define	VERSION		"2.8.2"
define	C		2.9979e5
define	LN2		0.69314718
define	CLN10		6.90292e5
define	TWOPI		6.283185

define  MIN_PIXEL_VALUE -1000000        # Minimum legal pixel value

define	MAX_RANGES	50
define	MAXTEMPS	2000
define	MAXCACHE	0
define	MAXLOG		5
#define MAXPTS		16384
#define MAXPTS2		8192
define	NPAR		4
define	SZ_HSTRING	68
define	SZ_HKWORD	9
 
define	CZMIN		-500.
define	CZMAX		50000.
define	SIGMIN		25.
define	SIGMAX		750.
define	GAMMIN		0.25
define	GAMMAX		1.50
define	CHIMIN		0.2
define	CHIMAX		10.0
define	ERRMIN		0.0001
define	ERRMAX		10.

#  Velocity correction flags
define	HC_VTYPES	"|none|file|heliocentric|barycentric|hfile|"
define  NONE		1	# no correction
define  FBCV		2	# Barycentric correction from file BCV
define	HCV		3	# Heliocentric correction
define	BCV		4	# Solar system barycentric correction
define  FHCV		5	# Heliocentric correction from file HCV

#  Velocity types
define  VCORREL		1		# Velocity from cross-correlation
define	VEMISS		2		# Velocity from emission line fit
define	VCOMB		3		# Velocity from emission and correlation
define	VSEARCH		4		# Velocity guessed from spectrum
define	VGUESS		5		# Velocity guess from parameter file
define	ZGUESS		6		# Velocity/c guess from parameter file
define	VCORTEMP	7		# Velocity from correlation of template

#  Velocity plotting flags
define	PL_VTYPES	"|correlation|emission|combination|search|guess|"

#  Initial velocity flags
define	IEM_VTYPES	"|correlation|emission|combination|search|guess|zguess|cortemp|"

# Correlation flags
define  COR_TYPES       "|no|velocity|pixel|wavelength|yes|"
define  COR_NO          1       # Do not cross-correlate
define  COR_VEL         2       # Cross-correlate velocity shift
define  COR_PIX         3       # Cross-correlate pixel shift
define  COR_WAV         4       # Cross-correlate wavelength shift
define  COR_YES         5       # Cross-correlate velocity shift

# Jul  3 1995	Last undocumented revision

# Feb  5 1997	Add velocity plotting and initial velocity flags

# Mar 20 1997	Version 2.0
# Apr 25 1997	Add zguess velocity flag
# May 19 1997	Add vcortemp velocity flat for initial velocities

# Feb 13 1998	Increase maximum number of correlation points to 16,384
# Mar  4 1998	Fix BCVCORR
# Apr  7 1998	Fix SUMSPEC
# May 15 1998	Allow minvel and maxvel to be INDEFD
# Jun 12 1998	2.0b17: Use pixel limits of WCS
# Jul 31 1998	2.0: Add toggling of heading in emsao mode 3 summary graph
# Dec 30 1998	2.1b1: Fix bug in XCSAO template pathname list handling
# Dec 30 1998	2.1b1: Increase maximum templates to 256

# Mar 18 1999	2.1b3: Multiply by 1000 after renormalizing
# Mar 30 1999	2.1b5: Allow use of OBSERVAT keyword for observatory location
# Apr  6 1999	2.1b6: Read templates only once
# May 11 1999	2.1b7: Fix SUMSPEC spectrum stacking
# Jun 10 1999	2.1b9: Add more options to SUMSPEC
# Jun 30 1999	2.1b11: Fix bug using template vel and HCV in XCSAO
# Jul 16 1999	2.1b12: Fix bug so QPLOT updates headers
# Jul 23 1999	2.1b13: Fix null report entries and add EMSAO tab table reports
# Jul 29 1999	2.1b14: Add XCSAO tab table reports; SUMSPEC constant normalize
# Aug 18 1999	2.1b15: SUMSPEC divide continuum no longer subtracts 1
# Sep  7 1999	2.1b16: Set CD matrix in SUMSPEC; fix dependency bugs
# Sep 15 1999	2.1b17: Fix tab table output from XCSAO
# Sep 24 1999	2.1.18: Fix EMSAO and XCSAO to update if quality flag is set
# Dec  6 1999	2.1.19: Free image WCS structure correctly

# Jan 25 2000	2.1.20: Add option to not remove object spectrum continuum
# Mar  6 2000	2.1.21: Fix bug in LINESPEC when using redshift as Z
# Apr 22 2000	2.1.22: Add 1900 to years from 0-120, not just 0-99
# Jul  5 2000	2.1.24: Automate renormalization
# Jul 21 2000	2.1.25: Add BCV adjustment without target velocity to SUMSPEC
# Aug  2 2000	2.1.26: Fix echelle spectrum number tracking
# Aug 10 2000	2.1.27: Fix bug in EMSAO which sorted lines poorly
# Sep 13 2000	Version 2.2: Add wavelength correlation and partial templates

# Feb  9 2001	2.2.1: Fix XCSAO to work with reversed spectra
# Mar 28 2001	2.2.2: Fix EMSAO to work with reversed spectra

# Feb  5 2002	2.2.5: Drop ends of spectra with values below MIN_PIXEL_VALUE
# Mar 29 2002	2.2.6: Add per spectrum wavelength limits to SUMSPEC
# May 29 2002	2.2.7: Add per spectrum velocity shifts to SUMSPEC
# Aug  7 2002	2.2.8: Add report mode 16 to XCSAO for per template wavelenghs
# Sep 17 2002	2.2.9: Allocate large vectors only once
# Sep 30 2002	2.3.0: Add task to compute equivalent widths including redshift

# Jun  2 2003	2.3.2: Add smoothing to SUMSPEC; fix BCVCORR
# Aug  1 2003	2.3.3: Add ability to read first extension of multi-ext. FITS

# Jan 16 2004	2.3.4: Print nonspec disp message only in debug mode in getimage()
# May 13 2004	2.3.5: Return fit from SUMSPEC; add forced emission labels to XCSAO
# Jul 20 2004	2.3.6: Read only first token from filename list files
# Aug  3 2004	2.3.7: Fix bug in SUMSPEC: use contin, not contout for cont rem.
# Sep 28 2004	2.3.8: Add parameters to EQWIDTH so it can be used on skies
# Oct 19 2004	2.3.9: Fix equivalent width computation in EMSAO so units match

# Jan 28 2005	2.4.0: Finally make equivalent width task EQWIDTH usable
# Mar 15 2005	2.4.1: Fix bug in RA setting in BCVCORR and compbcv()
# Mar 23 2005	2.4.2: Add constant graph scaling to SUMSPEC
# May 19 2005	2.4.3: Add pixel shift report mode, 3 decimal places to EMSAO
# Jul 14 2005:	2.4.4: Fix bugs for Linux port
# Jul 28 2005:	2.4.5: Make EQWIDTH output zeroes if band is off spectrum
# Aug 22 2005:	2.4.6: Print Z, ZEM, ZXC in header
# Sep 19 2005:	2.4.7: Fix bug in EQWIDTH flux computation
# Nov  3 2005:	2.4.7: Force renormalization in XCSAO if -1 < spectrum < 1
# Dec 20 2005:	2.4.8: Free all malloc'ed arrays in XCSAO

# Jan 12 2006:	2.4.8: Do not try to write file unless enabled by parameter
# May 31 2006:  2.5.0: Read name and position from APID, if present
# May 31 2006:  2.5.0: Add report mode 17 to xcsao for Hectochelle
# Jul 13 2006:	2.5.1: Do not use per-aperture keywords for apertures > 999
# Aug 17 2006:	2.5.2: Do not use zero-valued pixels at ends of spectrum
# Sep 27 2006:	2.5.3: Fix bug renormalizing very small numbers in XCSAO
# Oct 24 2006:	2.5.4: Increase max. number of templates to 512 from 256

# Jan 17 2007:	2.5.5: Plot all emission lines if any, not just those found
# Feb 14 2007	2.5.6: Preserve APNUMi in SUMSPEC from WCS strings
# Mar 30 2007	2.5.7: SUMSPEC exposure time addition and EMSAO reversed spectra
# Jun 13 2007	2.5.8: Read RA and DEC as strings not numbers from APIDi in getimage.x
# Jun 22 2007	2.5.9: Allocate transform vectors, so maximum number of bins is unfixed
# Aug  2 2007	2.6.0: Increase MAXTEMPS from 512 to 1024; drop MAXPTS, MAXPTS2
# Aug 13 2007	2.6.0: Increase MAXTEMPS to 2000
# Oct 29 2007	2.6.2: Save mean shift, err, R if echelle

# Feb 11 2008	2.6.3: Increase MAXTEMPS to 3000
# Mar  5 2008	2.6.4: Bad lines are in pixels if names start with "pix"
# May 13 2008	2.6.5: Add xplot and eplot to plot spectra without fitting
# May 28 2008	2.6.6: Add error computation and more report modes to EQWIDTH
# Nov 18 2008	2.6.9: Label emission lines with wavelengths in abs not labelled

# Jan  7 2009	2.7.0: Fix bug in SUMSPEC which put back BCV when it should not
# Jul 16 2009	2.7.7: Fix error in EQWIDTH and renormalize divided continuum
# Nov 20 2009	2.7.8: Increase file sizes from 256 to 511 characters

# Mar  1 2010	2.7.9: Write pixel shifts to image header in PEMSAO
# Jul  8 2010	2.7.9: Add FITS extension parameter mext to PXCSAO and PEMSAO

# Jun 27 2011	2.8.0: Use HJD or BJD from header even if no GJD

# May 25 2012	2.8.1: Add maxlength to PXCSAO and xcrslts.x for long file names
